client/src/api/getCourseInfo.ts

```
@@ -18,19 +19,21 @@ const getCourseInfo = async (
   year: string,
   term: string,
   courseCode: string,
-): Promise<CourseData | null> => {
+): Promise<CourseData | NetworkError> => {
   const baseURL = `${API_URL}/terms/${year}-${term}`;
   try {
     const data = await fetch(`${baseURL}/courses/${courseCode}/`);
+    if (data.status === 400) {
+      return { message: 'Internal server error' };
+    }
```

Instead of returning a error object, what do you think about throwing a custom error? In JS you can define customized errors like 
```javascript
class NetworkError extends Error {
  constructor(message = '') {
    super(message);
  }
}
```
and instead of `line 26`, we can write something like 
```javascript
if (data.status === 400) {
  throw new NetworkError('Internal server error');
}
```
 so that when you call `getCourseInfo()` you can do something like
```javascript
try {
  const selectedCourseClasses = await getCourseInfo(
    '2020',
    'T2',
    courseCode,
  );
  const newSelectedCourses = [...selectedCourses, selectedCourseClasses];
  populateTimetable(selectedCourseClasses); // TODO: temp until auto-timetabling is done
  setSelectedCourses(newSelectedCourses);
} catch (e) {
  if (e instanceof NetworkError) {
    setErrorMsg(selectedCourseClasses.message);
  }
}
```